### ISSUE TITLE
Multiline strings to support YAML format

### ISSUE THREAD

**Issue by ervuks** [jonsamwell/flutter_gherkin#111]
I have a use case where I would like to pass YAML in a multiline string and it would be converted to JSON (YAML is used for simplicity to write it). 
Example:
```
@multilineYaml
Feature: Multiline string YAML example
  Scenario: YAML scenario
    Given I can convert YAML to JSON structure
  """
  - name: YAML test
    type: simple_convertion
    value: 100
    conditions:
      - type: convertion
        value: Convertion value
  """
```
After multiline has been read, YAML is modified and the output now looks like this and from here it can't be correctly read with Dart: 
```
- name: YAML test
type: simple_convertion
value: 100
conditions:
- type: convertion
value: Convertion value
```

Point of code to check: 
https://github.com/jonsamwell/dart_gherkin/blob/45acf80426e94a4eb39f31cbe062ef26ea8ecde4/lib/src/gherkin/runnables/multi_line_string.dart#L23

&&

https://github.com/jonsamwell/dart_gherkin/blob/master/lib/src/gherkin/syntax/text_line_syntax.dart#L25

I was wondering if it's possible to put a flag or member property that would not perform trimming?

**Comment by jonsamwell** [comment]
Yes this is an interesting one, I trim it get rid of the formatting spaces most IDE's add when formatting multiline strings in feature files.  Do you know how this is handled in other gherkin implementations.  I am not sure how you would specify to not trim the string? I will have a think about this.

**Comment by ervuks** [comment]
In gherkin-java implementation, the closest I could find is https://github.com/cucumber/gherkin-java/blob/f8bb798cba46730877e45525d7f5433f5d47b461/src/main/java/io/cucumber/gherkin/GherkinLine.java#L43

And it follows rule from https://github.github.com/gfm/#info-string
Quote from the link above:
```
If the leading code fence is indented N spaces, then up to N spaces of indentation are removed from each line of the content (if present). (If a content line is not indented, it is preserved unchanged. If it is indented less than N spaces, all of the indentation is removed.)
```

**Comment by jonsamwell** [comment]
Yes that logic makes sense which would make the yaml still work I believe
if the number of leading spaces on the first line were removed from each
line.

I'll get that logic implemented

Thanks for you investigation!
